packages/core/src/api/apis/definitions/oauthrequest.ts

```diff
@@ -0,0 +1,173 @@
+ * before a popup is shown.
+ */
+export type AuthProviderInfo = {
+  /**
+   * Title for the auth provider, for example "GitHub"
+   */
+  title: string;
+
+  /**
+   * Icon for the auth provider.
+   */
+  icon: IconComponent;
+};
+
+/**
+ * Describes how to handle auth requests. Both how to show them to the user, and what to do when
+ * the user accesses the auth request.
+ */
+export type AuthRequesterOptions<T> = {
+  /**
+   * Information about the auth provider, which will be forwarded to auth requests.
+   */
+  info: AuthProviderInfo;
+
+  /**
+   * Implementation of the auth flow, which will be called synchronously when
+   * triggerAuth() is called on an auth requests.
+   */
+  authHandler(scope: OAuthScopes): Promise<T>;
+};
+
+/**
+ * Function used to trigger new auth requests for a set of scopes.
+ *
+ * The returned promise will resolve to the same value returned by the authHandler in the
+ * AuthRequesterOptions. Or rejected, if the request is rejected.
+ *
+ * This function can be called multiple times before the promise resolves. All calls
+ * will be merged into one request, and the scopes forwarded to the authHandler will be the
+ * union of all requested scopes.
+ */
+export type AuthRequester<T> = (scope: OAuthScopes) => Promise<T>;
+
+/**
+ * An auth request for a single auth provider.
+ */
+export type AuthRequest = {
+  /**
+   * Information about the auth provider, as given in the AuthRequesterOptions
+   */
+  info: AuthProviderInfo;
+
+  /**
+   * Rejects the request, causing all pending AuthRequester calls to fail with "RejectedError".
+   */
+  reject: () => void;
+
+  /**
+   * Synchronously calls the auth handler with all scope currently in the request.
+   */
+  triggerAuth(): Promise<void>;
+};
+
+/**
+ * Provides helpers for implemented OAuth login flows within Backstage.
+ */
+export type OAuthRequestApi = {
+  /**
+   * Show a popup pointing to a URL that starts an OAuth flow.
+   *
+   * The redirect handler of the flow should use postMessage to communicate back
+   * to the app window.
+   */
+  showLoginPopup(options: LoginPopupOptions): Promise<any>;
+
+  /**
+   * A utility for showing login popups or similar things, and merging together multiple requests for
+   * different scopes into one request that inclues all scopes.
+   *
+   * The passed in options provide information about the login provider, and how handle auth requests.
+   *
+   * The returned AuthRequester function is used to request login with new scopes. These requests
+   * are merged together and forwarded to the auth handler, as soon as a consumer of auth requests
+   * triggers an auth flow.
+   *
+   * See AuthRequesterOptions, AuthRequester, and handleAuthRequests for more info.
+   */
+  createAuthRequester<T>(options: AuthRequesterOptions<T>): AuthRequester<T>;
+
+  /**
+   * Subscribes as a handler of auth requests. The returned observable will emit all
```

Comment:
I mean it doesn't subscribe. You do that on the observable, right?

Reply:
Yep, dated docs